### ISSUE TITLE
Single stepping with CPU pos kernel

### ISSUE THREAD

**Issue by timspainUCL** [UCL/Glimpse#28]
Create a single stepping version (on a branch for now) of Glimpse that uses the existing `density_reconstruction::analysis_prox()` function. Get this to a state where it replicate the single stepping results. Once this is done, and the test version of Glimpse is running, re-write this as a test which can be merged into the main branch.

**Comment by timspainUCL** [comment]
The existing `density_reconstruction::analysis_prox()` does not give the same answer as the GPU code. There is additional complexity in gett ing the field into the form the function expects that is not present in the CUDA code.

Hence, I am writing the whole thing from scratch.

**Comment by timspainUCL** [comment]
![Comparison of CUDA and CPU results after 1 iteration](https://user-images.githubusercontent.com/51329178/66464301-a06c3f80-ea76-11e9-9b4f-84cb3d9cc376.png)

**Comment by timspainUCL** [comment]
![Screenshot 2019-10-18 at 09 44 51](https://user-images.githubusercontent.com/51329178/67080091-67c21980-f18c-11e9-8bf4-a08a97fd8419.png)

😀

**Comment by timspainUCL** [comment]
The change in density over the single iteration (δδ?) almost exactly the same between the CPU and CUDA code. So I'm calling this done!
The code can be found in the [`cpustep`](https://github.com/UCL/Glimpse/tree/cpustep) branch.
